app/scenes/Settings/Security.tsx

```diff
@@ -220,8 +247,62 @@ function Security() {
           short
         />
       </SettingRow>
+
+      <SettingRow
+        label={t("Allowed Domains")}
+        name="allowedDomains"
+        description={t(
+          "The domains which should be allowed to create accounts. This applies to both SSO and Email logins. Changing this setting does not affect existing user accounts."
+        )}
+      >
+        <div>
+          {data.allowedDomains &&
+            data.allowedDomains.map((domain, index) => (
+              <Flex key={index} gap={4}>
+                <Input
+                  key={index}
+                  id={`allowedDomains${index}`}
+                  value={domain}
+                  placeholder="example.com"
+                  onChange={(ev) => {
+                    const newData = { ...data };
+
+                    newData.allowedDomains[index] = ev.target.value;
+                    setData(newData);
+                  }}
+                />
+                <Remove>
+                  <Tooltip tooltip={t("Remove Domain")} placement="top">
+                    <NudeButton onClick={() => handleRemoveDomain(index)}>
+                      <CloseIcon />
+                    </NudeButton>
+                  </Tooltip>
+                </Remove>
+              </Flex>
+            ))}
+        </div>
+
+        <Flex justify="space-between" gap={4}>
+          {!data.allowedDomains?.length ||
+          data.allowedDomains[data.allowedDomains.length - 1] !== "" ? (
+            <Button type="button" onClick={handleAddDomain} neutral>
+              <Trans>Add another</Trans>
+            </Button>
+          ) : (
+            <span />
+          )}
+
+          <Button type="button" onClick={handleChange}>
+            <Trans>Save Domains</Trans>
+          </Button>
+        </Flex>
+      </SettingRow>
     </Scene>
   );
 }
 
+const Remove = styled("div")`
+  margin-top: 6px;
+`;
```

Comment:
This was copied from the Invite form

The `6px` seems like a magic number to just about center the icon. Didn't look pixel perfect to me but didn't futz with it too much. Let me know if you'd like me to!

Reply:
I think it's because inputs have a margin at the bottom but buttons don't so you can't flex align center. Seems okay.